data-prepper-plugins/mutate-event-processors/README.md

```diff
@@ -246,6 +246,101 @@ and the type conversion processor will change it to the following output, where
 * `key` - (required) - keys whose value needs to be converted to a different type
 * `type` - target type for the value of the key. Possible values are `integer`, `double`, `string`, and `boolean`. Default is `integer`.
 
+## List-to-map Processor
+A processor that converts a list of objects from an event, where each object has a key field, to a map of keys to objects.
+
+### Basic Usage
+
+To get started with list-to-map processor using Data Prepper, create the following `pipeline.yaml`:
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        value_key: "value"
+        flatten: true
+  sink:
+    - stdout:
+```
+
+Create the file named `logs_json.log` with the following content and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}]}
+```
+
+When run, the processor will parse the message into the following output:
+```json
+{"mylist":[{"name":"a","value":"val-a"},{"name":"b","value":"val-b1"},{"name":"b","value":"val-b2"},{"name":"c","value":"val-c"}],"a":"val-a","b":"val-b1","c":"val-c"}
```

Comment:
Prettier JSON please.

Reply:
This and those below were made prettier.